vortex-array/src/compute/is_constant.rs

```diff
@@ -35,14 +35,26 @@ pub fn is_constant(array: &dyn Array) -> VortexResult<Option<bool>> {
 ///
 /// Please see [`is_constant`] for a more detailed explanation of its behavior.
 pub fn is_constant_opts(array: &dyn Array, options: &IsConstantOpts) -> VortexResult<Option<bool>> {
-    Ok(IS_CONSTANT_FN
+    let result = IS_CONSTANT_FN
         .invoke(&InvocationArgs {
             inputs: &[array.into()],
             options,
         })?
         .unwrap_scalar()?
         .as_bool()
-        .value())
+        .value();
+
+    // TODO(joe): add is_constant for ListArray
+    if options.cost == Cost::Canonicalize && !array.is::<ListVTable>() {
```

Comment:
Move this logic inside the `impl ComputeFn for Constant`, in general we shouldn't assume callers will go through this function. They could go via FFI for example.